sklearn/manifold/t_sne.py

```diff
@@ -645,10 +645,10 @@ class TSNE(BaseEstimator):
     """
 
     def __init__(self, n_components=2, perplexity=30.0,
-                 early_exaggeration=4.0, learning_rate=1000.0, n_iter=1000,
+                 early_exaggeration=4.0, learning_rate=200.0, n_iter=2000,
```

Comment:
By the way, in the original pull request I wrote down some notes on the learning rate and compared several implementations ([link](https://github.com/scikit-learn/scikit-learn/pull/2822)):

> In the literature:
> - original paper: initialization with standard deviation 1e-4, 1000 episodes, learning rate 100, momentum 0.5 for 250 episodes, 0.8 for the rest, early exaggeration with 4 for 50 episodes
> - matlab implementation: learning rate 500, early exaggeration for 100 episodes
> - python implementation: initialization with standard deviation 1, learning rate 500, early exaggeration for 100 episodes, momentum 0.5 for 20 episodes
> - [divvy](https://github.com/jmlewis/divvy/blob/master/Plugins/Reducers/TSNE/tsne.c): initialization with standard deviation 1e-4, 1000 episodes, learning rate 1000, momentum 0.5 for 100 episodes, 0.8 for the rest, early exaggeration with 4 for 100 episodes
> - parametric t-sne (not comparable): conjugate gradient
> - barnes-hut t-sne: initialization with standard deviation 1e-4, 1000 episodes, learning rate 200, momentum 0.5 for 250 episodes, 0.8 for the rest, early exaggeration with 12 for 250 episodes
> 
> My experiences:
> - the learning rate has to be set manually for optimal performance, something between 100 and 1000
> - a high momentum (0.8) during early exaggeration improves the result
> 
> This implementation uses the following schedule:
> - initialization with standard deviation 1e-4, 1000 episodes, learning rate 1000, momentum 0.5 for 50 episodes, 0.8 for the rest, early exaggeration with 4 for 100 episodes

I wonder why we decided to use 1000 which is at the edge of the recommended range [100, 1000]?